zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessageCorrelationCorrelateProcessor.java

```diff
@@ -138,4 +160,42 @@ private MessageData createMessageData(
         messageCorrelationRecord.getVariablesBuffer(),
         messageCorrelationRecord.getTenantId());
   }
+
+  private Optional<Rejection> isAuthorizedForAllSubscriptions(
```

Comment:
🔧 This method can be refactored a bit:

```
private Optional<Rejection> checkAuthorizationForSubscriptions(
      final TypedRecord<MessageCorrelationRecord> command,
      final Subscriptions correlatingSubscriptions) {

    AuthorizationRequest authRequest = null;

    final var isAuthorized = correlatingSubscriptions.visitSubscriptions(
        subscription -> {

          PermissionType permission = subscription.isStartEventSubscription()
              ? PermissionType.CREATE
              : PermissionType.UPDATE;

          authRequest = new AuthorizationRequest(
              command, AuthorizationResourceType.PROCESS_DEFINITION, permission);
          authRequest.addResourceId(bufferAsString(subscription.getBpmnProcessId()));

          return authCheckBehavior.isAuthorized(authRequest);
        },
        true
    );

    if (!isAuthorized) {
      String errorMessage = UNAUTHORIZED_ERROR_MESSAGE.formatted(
          authRequest.getPermissionType(),
          authRequest.getResourceType()
      );
      return Optional.of(new Rejection(RejectionType.UNAUTHORIZED, errorMessage));
    }

    return Optional.empty();
  }
```

Reply:
That doesn't work. You cannot assign a value to `authRequest` from within a lambda.

Reply:
I'll take the permission type part though

Reply:
Oh yeah, sorry